### ISSUE TITLE
Nondeterministic ordering in generated schemas in PostgreSQL

### ISSUE THREAD

**Issue by stellingsimon** [jOOQ/jOOQ#7303]
### Expected behavior and actual behavior:

*Actual Behaviour:* We generate JOOQ-Files for various Schemas. We always run the JOOQ generator for two schemas at a time: a `shared` and one of the others, e.g. `contacts`. We observe that the ordering of schemas in the generated `DefaultCatalog.java` is not stable. Often (but not always) we end up with the following Diff after a generator run:

```
@@ -32,7 +32,7 @@ public class DefaultCatalog extends CatalogImpl {
     }
 
     private final List<Schema> getSchemas0() {
-        return Arrays.<Schema>asList(Catalogs.CONTACTS,
-                                     Shared.SHARED);
+        return Arrays.<Schema>asList(Shared.SHARED,
+                                     Catalogs.CONTACTS);
     }
 }
```

This is more than a nuisance since whenever we run the generation we have to manually undo these changes or else we risk merge conflicts. We are already post-processing the JOOQ files to guarantee a stable sorting of class members of all kinds, but since the diff above is inside an argument list, it rather hard for us to correct in a robust fashion.

It's unclear to me whether the changed ordering is due to JOOQ's, Postgres' or due to the sequence in which we invoke DB `create` statements, but since the order seems not to be relevant to the generated files, it should be stable IMHO.

*Expected Behaviour:* I'd like the schemas to be sorted by name to guarantee a stable sorting.

It seems to boil down to adding a `orderBy(PG_NAMESPACE.NSPNAME)` here for the Postgres case:

https://github.com/jOOQ/jOOQ/blob/b647b9167fa9656ce4c954c5ff5c5faa35fdae68/jOOQ-meta/src/main/java/org/jooq/util/postgres/PostgresDatabase.java#L518

### Steps to reproduce the problem:

Run the JOOQ generator on several schemas at once:

```
      <schemata>
        <schema>
          <inputSchema>shared</inputSchema>
        </schema>
        <schema>
          <inputSchema>contacts</inputSchema>
        </schema>
      </schemata>
```

### Versions:

- jOOQ: 3.10.5
- Java: 1.8
- Database (include vendor): Postgres 9.5
- OS: Linux
- JDBC Driver (include name if inofficial driver): Postgres JDBC 42.2.1, JDK8 edition

**Comment by lukaseder** [comment]
Thanks for reporting. Clearly a bug - all content produced by jOOQ-meta should be ordered by:

- Ordinal position (if available, e.g. columns, attributes, parameters)
- Qualified name

**Comment by lukaseder** [comment]
Check constraints and domain types are affected as well, and will be included in the fix

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.11.0. Will be merged to 3.10.6 (#7307, due later today) and 3.9.7 (#7308)
